compiler/luci/pass/src/CircleQuantizer.test.cpp

```diff
@@ -189,3 +193,240 @@ TEST(CircleQuantizerTest, quantize_requant_output_NEG)
+  options->param(AlgorithmParameters::Quantize_granularity, "layer");
+  return options;
+}
+
+using LayerParam = luci::CircleQuantizer::Options::LayerParam;
+using LayerParams = luci::CircleQuantizer::Options::LayerParams;
+using LayerParamsSet = luci::CircleQuantizer::Options::LayerParamsSet;
+
+TEST(CircleQuantizerTest, quantize_layer_param_set)
+{
+  SimpleQuantGraph sqg;
+  sqg.init();
+
+  LayerParamsSet lpss;
+  {
+    LayerParams lps1;
+    {
+      auto lp1 = std::make_shared<LayerParam>();
+      lp1->name = "x1";
+      lp1->dtype = "int16";
+      lp1->granularity = "channel";
+      lps1.emplace_back(lp1);
+    }
+    lpss.emplace_back(lps1);
+
+    LayerParams lps2;
+    {
+      auto lp2 = std::make_shared<LayerParam>();
+      lp2->name = "c1";
+      lp2->dtype = "int16";
+      lp2->granularity = "channel";
+      lps2.emplace_back(lp2);
+    }
+    lpss.emplace_back(lps2);
+  }
+
+  SimpleCircleQuantizer scq;
+  auto options = scq.init();
+  options->layer_params_set(lpss);
+
+  EXPECT_NO_THROW(scq.quantize(&sqg.g));
+}
+
+TEST(CircleQuantizerTest, quantize_layer_param_set_no_NEG)
+{
+  SimpleQuantGraph sqg;
+  sqg.init();
+
+  LayerParamsSet lpss;
+  {
+    // there is no LayerParam with "c1" nor "c2"
+    LayerParams lps1;
+    {
+      auto lp1 = std::make_shared<LayerParam>();
+      lp1->name = "x1";
+      lp1->dtype = "int16";
+      lp1->granularity = "channel";
+      lps1.emplace_back(lp1);
+    }
+    lpss.emplace_back(lps1);
+
+    LayerParams lps2;
+    {
+      auto lp2 = std::make_shared<LayerParam>();
+      lp2->name = "x2";
+      lp2->dtype = "int16";
+      lp2->granularity = "channel";
+      lps2.emplace_back(lp2);
+    }
+    lpss.emplace_back(lps2);
+  }
+
+  SimpleCircleQuantizer scq;
+  auto options = scq.init();
+  options->layer_params_set(lpss);
+
+  EXPECT_THROW(scq.quantize(&sqg.g), std::runtime_error);
+}
+
+TEST(CircleQuantizerTest, quantize_layer_param_set_dup1_NEG)
+{
+  SimpleQuantGraph sqg;
+  sqg.init();
+
+  LayerParamsSet lpss;
+  {
+    LayerParams lps1;
+    {
+      // duplicate c1 name in a LayerParams
+      auto lp11 = std::make_shared<LayerParam>();
+      lp11->name = "c1";
+      lp11->dtype = "int16";
+      lp11->granularity = "channel";
+      lps1.emplace_back(lp11);
+
+      auto lp12 = std::make_shared<LayerParam>();
+      lp12->name = "c1";
+      lp12->dtype = "int16";
+      lp12->granularity = "channel";
+      lps1.emplace_back(lp12);
+    }
+    lpss.emplace_back(lps1);
+
+    LayerParams lps2;
+    {
+      auto lp2 = std::make_shared<LayerParam>();
+      lp2->name = "x1";
+      lp2->dtype = "int16";
+      lp2->granularity = "channel";
+      lps2.emplace_back(lp2);
+    }
+    lpss.emplace_back(lps2);
+  }
+
+  SimpleCircleQuantizer scq;
+  auto options = scq.init();
+  options->layer_params_set(lpss);
+
+  EXPECT_THROW(scq.quantize(&sqg.g), std::runtime_error);
+}
+
+TEST(CircleQuantizerTest, quantize_layer_param_set_dup2_NEG)
```

Comment:
IIUC, this tests the case for multiple valid `LayerParams`.

```suggestion
TEST(CircleQuantizerTest, multiple_valid_layer_param_set_NEG)
```

Reply:
ah, `valid` may give confusion. how about just `multiple_layer_param_set_NEG` ?

Reply:
To describe why this is a negative test, how about `duplicate_layer_params_NEG `?